packages/js/src/executors/swc/swc.impl.ts

```diff
@@ -63,52 +65,71 @@ export function normalizeOptions(
 }
 
 export async function* swcExecutor(
-  options: SwcExecutorOptions,
+  _options: SwcExecutorOptions,
   context: ExecutorContext
 ) {
   const { sourceRoot, root } = context.workspace.projects[context.projectName];
-  const normalizedOptions = normalizeOptions(
-    options,
-    context.root,
-    sourceRoot,
-    root
-  );
-  normalizedOptions.swcrcPath = addTempSwcrc(normalizedOptions);
+  const options = normalizeOptions(_options, context.root, sourceRoot, root);
+  options.swcrcPath = addTempSwcrc(options);
   const { tmpTsConfig, projectRoot } = checkDependencies(
     context,
     options.tsConfig
   );
 
   if (tmpTsConfig) {
-    normalizedOptions.tsConfig = tmpTsConfig;
+    options.tsConfig = tmpTsConfig;
+  }
+
+  const assetHandler = new CopyAssetsHandler({
+    projectDir: projectRoot,
+    rootDir: context.root,
+    outputDir: options.outputPath,
+    assets: options.assets,
+  });
+
+  if (options.watch) {
+    const disposeWatchAssetChanges =
+      await assetHandler.watchAndProcessOnAssetChange();
+    const disposeReadmeChanges = await watchForSingleFileChanges(
+      join(context.root, projectRoot),
+      'package.json',
+      () =>
+        updatePackageJson(
+          options.main,
+          options.outputPath,
+          projectRoot,
+          !options.skipTypeCheck
+        )
+    );
```

Comment:
This watches for package JSON changes, not readme changes. Probably intended, but variable name should be refactored.